src/Compilers/Core/Portable/Diagnostic/Diagnostic.cs

```diff
@@ -15,7 +15,7 @@ namespace Microsoft.CodeAnalysis
     /// Represents a diagnostic, such as a compiler error or a warning, along with the location where it occurred.
     /// </summary>
     [DebuggerDisplay("{GetDebuggerDisplay(), nq}")]
-    public abstract partial class Diagnostic : IEquatable<Diagnostic>, IFormattable
```

Comment:
Yes. I could leave ```IEquatable<Diagnostic>``` but that seems to imply more than ```ReferenceEquals```. Or we could have ```Equals``` implement a deeper compare, perhaps comparing string representations. Unfortunately, the previous Equals relied on the individual Arguments implementing Equals which is not always the case: see #6262.